### ISSUE TITLE
Add ability to annotate terminators. 

### ISSUE THREAD

**Issue by monicacecilia** [GMOD/Apollo#207]
As suggested by @erasche, it would be nice to have the ability to add other genomic features such as terminators. 

We will welcome help in defining the elements that can be represented: 
* Will we need to distinguish between Rho-dependent and Rho-independent terminators? 
* More information on how these elements are identified / predicted and additional details on their recognition - how does RNApol II recognizes them?

**Comment by erasche** [comment]
> Will we need to distinguish between Rho-dependent and Rho-independent terminators?

yes

> More information on how these elements are identified / predicted and additional details on their recognition - how does RNApol II recognizes them?

- some info under principles, http://rna.igmors.u-psud.fr/toolbox/arnold/
- RNAmotif paper http://nar.oxfordjournals.org/content/29/17/3583.full
- rho dependent are discoverable with wet lab experiments http://www.ncbi.nlm.nih.gov/pubmed/16946247

**Comment by erasche** [comment]
@monicacecilia, my PI is nagging me, do you need any more information for this?

**Comment by monicacecilia** [comment]
Dear @erasche I'm sorry you are being nagged. 

We do not currently require additional information. 

We are currently planning to start working on this for Milestone 2.2.0. 

If this is a pressing matter that your lab needs to address, please feel free to start looking into it yourself, and we can guide you through the sticky parts. Unfortunately for the purpose of your progress, we have 2 other, very time-consuming priorities ahead of everything else (annotation of variants and transformation of coordinates).  I wish we had better news for you at the moment.

**Comment by erasche** [comment]
No worries @monicacecilia, was just following up since the issue hadn't been touched in ~6 months. Completely understandable with respect to higher priority items. Thanks for the response!

**Comment by monicacecilia** [comment]
Dear @erasche, would you settle for an additional feature type from the list? 
And the ability to identify them on the "User-created Annotation" area? 
You would not yet be able to calculate the effect of the terminator on the transcript.

**Comment by erasche** [comment]
@monicacecilia yes! We don't need to calculate anything, just to make the annotation. I'll look into where that lives in the codebase.

**Comment by monicacecilia** [comment]
Brilliant!

**Comment by monicacecilia** [comment]
- [ ] Add feature type 'terminator'. Seen here in [Sequence Ontology](http://www.sequenceontology.org/miso/current_svn/term/SO:0000141). 

- Note subtypes: bacterial (rho-dependent and rho-independent) and eukaryotic (terminator of RNA Pol-II promoter).

**Comment by nathandunn** [comment]
We were discussing and we think that [promoter](http://www.sequenceontology.org/browser/release_2.3/term/SO:0000167) should be added as well, but we can add that later (it should follow the exact same pattern).  

They can probably both mirror the [Transcript](https://github.com/GMOD/Apollo/blob/master/grails-app/domain/org/bbop/apollo/Transcript.groovy) domain object in by extending the GeneMemberRegion (but with custom terms, SO, etc.), but mimic RepeatRegion and Tranposable Element in terms of functionality.     I think you're familiar with [AnnotTrack.js](https://github.com/GMOD/Apollo/blob/master/client/apollo/js/View/Track/AnnotTrack.js#L4524) but the pattern is similar there.   I'd suggest a tool like IntelliJ to work with grails.

**Comment by nathandunn** [comment]
I would set the property Rho-depdenent property directly on the object.
